### DISCUSSION TITLE
Add the local crate failed.

### DISCUSSION THREAD

**Discussion by baiguoname** [sigmaSd/IRust#111] [Q&A]
I am using `irust_repl` crate like this:
`use irust_repl::{EvalConfig, Repl, DEFAULT_EVALUATOR};

fn main() {
    let mut repl = Repl::default();
    repl.add_dep(&["chrono".to_string(), "rand".to_string(), "serde".to_string()]).unwrap().wait().unwrap();
    repl.insert(
        r#"
        use chrono::NaiveDateTime;
        let dt = NaiveDateTime::default();
        println!("{}", dt);
}`

It works fine. But when I add local crate, it fails. What am I doing wrong?
`
    let mut repl = Repl::default();
    repl.add_dep(&[r#"name = { path = "local_crate_path" }"#.to_string()]).unwrap().wait().unwrap();
    repl.insert(
        r#"
        use chrono::NaiveDateTime;
        let dt = NaiveDateTime::default();
        println!("{}", dt);
`

**Comment by baiguoname** [comment]
The code is hard to read, so I paste a picture:
![image](https://user-images.githubusercontent.com/52092777/230265951-4eea2f8e-25fb-49a3-9a53-665ad8120b0a.png)

**Comment by sigmaSd** [comment]
add_dep needs to be used exactly like use cargo add cli, so `repl.add_dep(&["--path".into(), "/tmp/lab/hello".into()])`
and the path needs to be an absolute path

for example
```rs
use irust_repl::Repl;

fn main() {
    let mut repl = Repl::default();
    repl.add_dep(&["--path".into(), "/tmp/lab/hello".into()])
        .unwrap()
        .wait()
        .unwrap();
    dbg!(repl.eval(r#"println!("{}",hello::add(1,2))"#).unwrap());
}
```
